discovery_engine_core/providers/src/client.rs

```diff
@@ -46,383 +49,141 @@ pub enum Error {
-        }
+    /// How much the article match the query.
+    pub score: Option<f32>,
 
-        Ok(())
-    }
-}
+    /// The country of the publisher.
+    pub country: String,
 
-/// Parameters determining which news to fetch
-pub struct NewsQuery<'a, F> {
-    /// Common parts
-    pub common: CommonQueryParts<'a>,
-    /// News filter.
-    pub filter: F,
+    /// The language of the article.
+    pub language: String,
+
+    /// Main topic of the publisher.
+    pub topic: String,
 }
 
-impl<F> Query for NewsQuery<'_, F>
-where
-    F: Deref<Target = Filter> + Sync,
-{
-    fn setup_url(&self, url: &mut Url) -> Result<(), Error> {
-        self.common.setup_url(url, "_sn")?;
+impl Article {
+    /// Gets the excerpt or falls back to the title if the excerpt is empty.
+    pub fn snippet_or_title(&self) -> &str {
+        (!self.snippet.is_empty())
+            .then(|| &self.snippet)
+            .unwrap_or(&self.title)
+    }
 
-        let mut query = url.query_pairs_mut();
-        query
-            .append_pair("sort_by", "relevancy")
-            .append_pair("q", &self.filter.build());
+    /// Loads newscatcher articles from JSON and returns them in `Article` representation
+    pub fn load_from_newscatcher_json_representation(json: &str) -> Result<Vec<Self>, Error> {
+        let articles: Vec<NewscatcherArticle> =
+            serde_json::from_str(json).map_err(Error::Decoding)?;
 
-        Ok(())
+        Ok(articles.into_iter().map(|article| article.into()).collect())
     }
 }
 
-impl<T> Seal for NewsQuery<'_, T> {}
-
-/// Parameters determining which headlines to fetch.
-pub struct HeadlinesQuery<'a> {
-    /// Common parts.
-    pub common: CommonQueryParts<'a>,
-    /// Trusted sources.
-    pub trusted_sources: &'a [String],
-    /// Headlines topic.
-    pub topic: Option<&'a str>,
+impl From<NewscatcherArticle> for Article {
+    fn from(source: NewscatcherArticle) -> Self {
+        Article {
+            title: source.title,
+            snippet: source.excerpt,
+            url: source.link,
+            source_domain: source.source_domain,
+            date_published: source.published_date,
+            image: source.media,
+            rank: source.rank,
+            score: source.score,
+            country: source.country,
+            language: source.language,
+            topic: source.topic.to_string(),
+        }
+    }
 }
 
-impl Query for HeadlinesQuery<'_> {
-    fn setup_url(&self, url: &mut Url) -> Result<(), Error> {
-        self.common.setup_url(url, "_lh")?;
-
-        let mut query = url.query_pairs_mut();
-        if !self.trusted_sources.is_empty() {
-            query.append_pair("sources", &self.trusted_sources.join(","));
-        };
-        if let Some(topic) = self.topic {
-            query.append_pair("topic", topic);
+impl From<GnewsArticle> for Article {
+    fn from(source: GnewsArticle) -> Self {
+        let source_domain = Url::parse(&source.url)
```

Comment:
Not sure if we need better handling for this. If the article URL is invalid, returning an empty source domain isn't optimal. However, our other filtering steps should definitely catch articles with invalid URLs, so this article should be filtered out down the line anyway 🤔